x/liquidity/amm/pool.go

```diff
@@ -151,20 +367,43 @@ func (pool *BasicPool) BuyAmountOver(price sdk.Dec) (amt sdk.Int) {
 	return
 }
 
-// SellAmountUnder returns the amount of sell orders for price less than
-// or equal to given price.
-func (pool *BasicPool) SellAmountUnder(price sdk.Dec) sdk.Int {
-	return pool.ProvidableYAmountUnder(price)
+func (pool *RangedPool) SellAmountUnder(price sdk.Dec, _ bool) (amt sdk.Int) {
+	if price.GT(pool.maxPrice) {
+		price = pool.maxPrice
+	}
+	if price.LTE(pool.Price()) {
+		return zeroInt
+	}
+	// dy = (ry + transY) - (rx + transX) / P
+	amt = pool.yComp.Sub(pool.xComp.QuoRoundUp(price)).TruncateInt()
```

Comment:
`amt` is the amount that the pool gives. Then, should we use `QuoTruncate` rather than `QuoRoundUp`?